utxo.go

```diff
@@ -0,0 +1,192 @@
+	var outputs = make([]*wire.TxOut, 0)
+	var totalSendAmount int64
+	var maxAmountRecipientAddress string
+
+	for _, destination := range txDestinations {
+		// validate the amount to send to this destination address
+		if !destination.SendMax && (destination.AtomAmount <= 0 || destination.AtomAmount > dcrutil.MaxAmount) {
+			return nil, 0, "", errors.E(errors.Invalid, "invalid amount")
+		}
+
+		// check if multiple destinations are set to receive max amount
+		if destination.SendMax && maxAmountRecipientAddress != "" {
+			return nil, 0, "", fmt.Errorf("cannot send max amount to multiple recipients")
+		}
+
+		if destination.SendMax {
+			maxAmountRecipientAddress = destination.Address
+			continue // do not prepare a tx output for this destination
+		}
+
+		output, err := txhelper.MakeTxOutput(destination.Address, destination.AtomAmount, tx.sourceWallet.chainParams)
+		if err != nil {
+			return nil, 0, "", fmt.Errorf("make tx output error: %v", err)
+		}
+
+		totalSendAmount += output.Value
+		outputs = append(outputs, output)
+	}
+
+	return outputs, totalSendAmount, maxAmountRecipientAddress, nil
+}
+
+func (tx *TxAuthor) constructCustomTransaction() (*txauthor.AuthoredTx, error) {
+	// Used to generate an internal address for change,
+	// if no change destination is provided and
+	// no recipient is set to receive max amount.
+	nextInternalAddress := func() (string, error) {
+		ctx := tx.sourceWallet.shutdownContext()
+		addr, err := tx.sourceWallet.internal.NewChangeAddress(ctx, tx.sourceAccountNumber)
+		if err != nil {
+			return "", err
+		}
+		return addr.Address(), nil
+	}
+
+	return tx.newUnsignedTxUTXO(tx.inputs, tx.destinations, tx.changeDestinations, nextInternalAddress)
+}
+
+func (tx *TxAuthor) newUnsignedTxUTXO(inputs []*wire.TxIn, sendDestinations, changeDestinations []TransactionDestination,
+	nextInternalAddress nextAddressFunc) (*txauthor.AuthoredTx, error) {
+	outputs, totalSendAmount, maxAmountRecipientAddress, err := tx.ParseOutputsAndChangeDestination(sendDestinations)
+	if err != nil {
+		return nil, err
+	}
+
+	if maxAmountRecipientAddress != "" && len(changeDestinations) > 0 {
+		return nil, errors.E(errors.Invalid, "no change is generated when sending max amount,"+
+			" change destinations must not be provided")
+	}
+
+	if maxAmountRecipientAddress == "" {
+		// no change specified, generate new internal address to use as change (max amount recipient)
+		maxAmountRecipientAddress, err = nextInternalAddress()
+		if err != nil {
+			return nil, fmt.Errorf("error generating internal address to use as change: %s", err.Error())
+		}
+	}
+
+	var totalInputAmount int64
+	inputScriptSizes := make([]int, len(inputs))
+	inputScripts := make([][]byte, len(inputs))
+	for i, input := range inputs {
+		totalInputAmount += input.ValueIn
+		inputScriptSizes[i] = txsizes.RedeemP2PKHSigScriptSize
+		inputScripts[i] = input.SignatureScript
+	}
+
+	var changeScriptSize int
+	if maxAmountRecipientAddress != "" {
```

Comment:
This will always evaluate to true because you assigned it here:
https://github.com/planetdecred/dcrlibwallet/blob/728a131272971ba04eb89b47736ff84d99a24fdb/utxo.go#L107-L113

I'd suggest you replace:
https://github.com/planetdecred/dcrlibwallet/blob/728a131272971ba04eb89b47736ff84d99a24fdb/utxo.go#L107

With:`if maxAmountRecipientAddress == "" && len(changeDestinations) == 0 {` to make this logic work.